internal/kafka/internal/services/clusters.go

```diff
@@ -665,3 +668,120 @@ func (c clusterService) IsStrimziKafkaVersionAvailableInCluster(cluster *api.Clu
+	for _, KafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
+		if KafkaStreamingUnitCountPerCluster.ClusterId == clusterId && KafkaStreamingUnitCountPerCluster.InstanceType == instanceType {
+			return int(KafkaStreamingUnitCountPerCluster.Count)
+		}
+	}
+
+	return 0
+}
+
+// KafkaPerClusterCount is a struct used to query the database using a "group by" clause
+type KafkaPerClusterCount struct {
+	Region        string
+	InstanceType  string
+	ClusterId     string
+	Count         int32
+	CloudProvider string
+	SizeId        string
+}
+
+type ClusterSelection struct {
+	CloudProvider         string
+	ClusterID             string
+	Region                string
+	SupportedInstanceType string
+	DynamicCapacityInfo   api.JSON
+}
+
+func (c *clusterService) FindStreamingUnitCountByClusterAndInstanceType() (KafkaStreamingUnitCountPerClusterList, error) {
+
+	var clusters []*ClusterSelection
+	dbConn := c.connectionFactory.New().
+		Model(&api.Cluster{}).
+		Where("cluster_id != ''")
+
+	if err := dbConn.Scan(&clusters).Error; err != nil {
+		return nil, apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to list clusters")
+	}
+
+	streamingUnitsCountPerRegion := KafkaStreamingUnitCountPerClusterList{}
+
+	// pre-populate regions count with zero count values.
+	// This is useful and it ensures that the count drops to 0 for each instance type and region when all the kafkas are removed
+	for _, clusterSelection := range clusters {
+		supportedInstanceTypes := strings.Split(clusterSelection.SupportedInstanceType, ",")
+
+		c := &api.Cluster{DynamicCapacityInfo: clusterSelection.DynamicCapacityInfo}
+		clusterDynamicCapacityInfo, _ := c.RetrieveDynamicCapacityInfo()
+
+		for _, supportedInstanceType := range supportedInstanceTypes {
+			instanceType := strings.TrimSpace(supportedInstanceType)
+			if instanceType == "" {
+				continue
+			}
+
+			var maxUnits int32 = 0
+			instanceCapacityInfo, ok := clusterDynamicCapacityInfo[instanceType]
+			if ok {
+				maxUnits = instanceCapacityInfo.MaxUnits
+			}
+
+			streamingUnitsCountPerRegion = append(streamingUnitsCountPerRegion, KafkaStreamingUnitCountPerCluster{
+				CloudProvider: clusterSelection.CloudProvider,
+				ClusterId:     clusterSelection.ClusterID,
+				InstanceType:  instanceType,
+				Region:        clusterSelection.Region,
+				Count:         0,
+				MaxUnits:      maxUnits,
+			})
+		}
+	}
+
+	dbConn = c.connectionFactory.New()
+	var kafkasPerRegion []*KafkaPerClusterCount
+	if err := dbConn.Model(&dbapi.KafkaRequest{}).
+		Select("cloud_provider, region, count(1) as Count, size_id, cluster_id, instance_type").
+		Group("size_id, cluster_id, cloud_provider, region, instance_type").
+		Scan(&kafkasPerRegion).Error; err != nil {
+		return nil, apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to perform count query on kafkas table")
+	}
+
+	for _, kafkaCountPerRegion := range kafkasPerRegion {
```

Comment:
Wouldn't the name of the iterator variable be `kafkaCountPerCluster`? I understand each element of `kafkasPerRegion` would be a struct containing the count of kafkas in a specific cluster in a region